src/taipy/core/data/csv.py

```diff
@@ -82,17 +85,20 @@ def __init__(
         owner_id: Optional[str] = None,
         parent_ids: Optional[Set[str]] = None,
         last_edit_date: Optional[datetime] = None,
-        edits: List[Edit] = None,
-        version: str = None,
+        edits: Optional[List[Edit]] = None,
+        version: Optional[str] = None,
         validity_period: Optional[timedelta] = None,
         edit_in_progress: bool = False,
-        properties: Dict = None,
+        properties: Optional[Dict] = None,
     ):
         if properties is None:
             properties = {}
 
         default_value = properties.pop(self.__DEFAULT_DATA_KEY, None)
 
+        if self.__ENCODING_KEY not in properties.keys():
+            properties[self.__ENCODING_KEY] = "utf-8"
```

Comment:
I was thinking about it, but we do not use any kind of `__HAS_HEADER_DEFAULT` or something like that at the DataNode level.

I think it is because this is just safety layer, because a DataNode should be created from a Configuration, and the default value is set at the config already.

If we use `__ENCODING_DEFAULT`, we should do the same for other parameter as well. Let me know what you think